src/main/webapp/personalised-recommendations.html

```diff
@@ -26,46 +26,80 @@ <h1 class="title">Personalised Recommendations</h1>
-                <label class="flex-item"><input type="radio" value="2" name="crowd"></label>
-                <label class="flex-item"><input type="radio" value="3" name="crowd"></label>
-                <label class="flex-item"><input type="radio" value="4" name="crowd"></label>
-                <label class="flex-item"><input type="radio" value="5" name="crowd"></label>
-                <div class="flex-item">Loud</div>
-            </div>
-            <button class="button">Generate Recommendations</button>
+        <div class="preferences">
+          <h2>Preferences</h2>
+          <div class="button-row">
+            <label class="category active">
+              <input type="radio" name="recommendation-category" id="restaurants"> Restaurants
+            </label>
+            <label class="category">
+              <input type="radio" name="recommendation-category" id="places-to-visit"> Places to Visit
+            </label>
+            <label class="category">
+              <input type="radio" name="recommendation-category" id="bars-clubs"> Bars & Clubs
+            </label>
+            <label class="category">
+              <input type="radio" name="recommendation-category" id="study-places"> Study Places
+            </label>
           </div>
+          <div class="button-row">
+              <div class="flex-item">Cheap</div>
+              <label class="flex-item"><input type="radio" value="1" name="price"></label>
+              <label class="flex-item"><input type="radio" value="2" name="price"></label>
+              <label class="flex-item"><input type="radio" value="3" name="price"></label>
+              <label class="flex-item"><input type="radio" value="4" name="price"></label>
+              <label class="flex-item"><input type="radio" value="5" name="price"></label>
+              <div class="flex-item">Expensive</div>
+          </div>
+          <div class="button-row">
+              <div class="flex-item">Quiet</div>
+              <label class="flex-item"><input type="radio" value="1" name="crowd"></label>
+              <label class="flex-item"><input type="radio" value="2" name="crowd"></label>
+              <label class="flex-item"><input type="radio" value="3" name="crowd"></label>
+              <label class="flex-item"><input type="radio" value="4" name="crowd"></label>
+              <label class="flex-item"><input type="radio" value="5" name="crowd"></label>
+              <div class="flex-item">Loud</div>
+          </div>
+          <button class="button">Generate Recommendations</button>
+        </div>
+      </div>
+      <!-- Displaying top 5 recommendations -->
+      <div class="recommendations-container">
+        <h2>Top 5 Recommendations</h2>
+        <div class="recommendation-box">
+          <!-- Sample text to show where the recommendation format -->
+          <p id="top-recommendation">Most Recommended</p>
```

Comment:
since we are going to fill up the 5 recommendation box from the server and script, we will need a way to differ the boxes, thus might have to change the ids. However this might be kinda tricky to fill up from the script.
An alternative I am thinking about would be to do 5 times iteration and create the recommendation boxes from script, i.e. the pseudocode will be
for i: 1 to 5 {
  create div recommendation-box
  recommendation-box.append(category-header)
  recommendation-box.append(place-title)
  etc
}
feel free to address this comment now or later since I see you are focusing on the HTML for now

Reply:
I'll update the IDs to be unique for now, as that will make future implementation easier, but yes, we can also consider iterating through and creating the boxes. This will be the next step anyway - implementing the actual 'dynamic' HTML based on the algorithm.